skimage/feature/_hog.py

```diff
@@ -97,19 +124,34 @@ def hog(image, orientations=9, pixels_per_cell=(8, 8),
     e.g. bar like structures in bicycles and limbs in humans.
     """
 
+
     if image.dtype.kind == 'u':
         # convert uint image to float
         # to avoid problems with subtracting unsigned numbers in np.diff()
         image = image.astype('float')
 
-    gx = np.empty(image.shape, dtype=np.double)
-    gx[:, 0] = 0
-    gx[:, -1] = 0
-    gx[:, 1:-1] = image[:, 2:] - image[:, :-2]
-    gy = np.empty(image.shape, dtype=np.double)
-    gy[0, :] = 0
-    gy[-1, :] = 0
-    gy[1:-1, :] = image[2:, :] - image[:-2, :]
+    if len(image.shape) == 3:
+
+        channel_gradients = []
+        gradient = np.zeros(image.shape)
+
+        for dimension in range(image.shape[2]):
+            gx, gy = compute_image_gradients(image[:, :, dimension])
+            gradient[:, :, dimension] = np.hypot(gx, gy)
+            channel_gradients.append((gx, gy))
+
+        gx = np.empty(image.shape[0:2], dtype=np.double)
+        gy = np.empty(image.shape[0:2], dtype=np.double)
+
+        max_index_array = gradient.argmax(axis = 2)
+
+        for row_index, row in enumerate(max_index_array):
+            for col_index, value in enumerate(row):
+                gx[row_index][col_index] = channel_gradients[value][0][row_index][col_index]
```

Comment:
Could you wrap this to 80 chars ?

Please follow PEP8 styling guidelines.
You can use `flake8` or `autopep8` for help